R/opts.R

```diff
@@ -0,0 +1,690 @@
+#' serves as a shortcut for using `tab_options(footnotes.marks = {marks})`
+#'
+#' We can supply a vector of that will represent the series of marks.
+#' The series of footnote marks is recycled when its usage goes beyond the
+#' length of the set. At each cycle, the marks are simply doubled, tripled, and
+#' so on (e.g., `*` -> `**` -> `***`). The option exists for providing keywords
+#' for certain types of footnote marks. The keywords are:
+#'
+#' \itemize{
+#' \item `"numbers"`: numeric marks, they begin from 1 and these marks are not
+#' subject to recycling behavior
+#' \item `"letters"`: miniscule alphabetic marks, internally uses the `letters`
+#' vector
+#' which contains 26 lowercase letters of the Roman alphabet
+#' \item `"LETTERS"`: majuscule alphabetic marks, using the `LETTERS` vector
+#' which has 26 uppercase letters of the Roman alphabet
+#' \item `"standard"`: symbolic marks, four symbols in total
+#' \item `"extended"`: symbolic marks, extends the standard set by adding two
+#' more symbols, making six
+#' }
+#'
+#' @inheritParams fmt_number
+#' @param marks Either a vector (that will represent the series of marks) or a
+#'   keyword that represents a preset sequence of marks. The valid keywords are:
+#'   `"numbers"` (for numeric marks), `"letters"` and `"LETTERS"` (for lowercase
+#'   and uppercase alphabetic marks), `"standard"` (for a traditional set of
+#'   four symbol marks), and `"extended"` (which adds two more symbols to the
+#'   standard set).
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @examples
+#' # Use `sza` to create a gt table,
+#' # adding three footnotes; call
+#' # `opt_footnote_marks()` to specify
+#' # which footnote marks to use
+#' tab_1 <-
+#'   sza %>%
+#'   dplyr::group_by(latitude, tst) %>%
+#'   dplyr::summarize(
+#'     SZA.Max = max(sza),
+#'     SZA.Min = min(sza, na.rm = TRUE)
+#'   ) %>%
+#'   dplyr::ungroup() %>%
+#'   dplyr::filter(latitude == 30, !is.infinite(SZA.Min)) %>%
+#'   dplyr::select(-latitude) %>%
+#'   gt(rowname_col = "tst") %>%
+#'   tab_spanner_delim(delim = ".") %>%
+#'   fmt_missing(
+#'     columns = everything(),
+#'     missing_text = "90+"
+#'   ) %>%
+#'   tab_stubhead("TST") %>%
+#'   tab_footnote(
+#'     footnote = "True solar time.",
+#'     locations = cells_stubhead()
+#'   ) %>%
+#'   tab_footnote(
+#'     footnote = "Solar zenith angle.",
+#'     locations = cells_column_spanners(spanners = "SZA")
+#'   ) %>%
+#'   tab_footnote(
+#'     footnote = "The Lowest SZA.",
+#'     locations = cells_stub(rows = "1200")
+#'   ) %>%
+#'   opt_footnote_marks(marks = "standard")
+#'
+#' @section Figures:
+#' \if{html}{\figure{man_opt_footnote_marks_1.svg}{options: width=100\%}}
+#'
+#' @family Table Option Functions
+#' @section Function ID:
+#' 9-1
+#'
+#' @export
+opt_footnote_marks <- function(data,
+                               marks = NULL) {
+
+  # Perform input object validation
+  stop_if_not_gt(data = data)
+
```

Comment:
Validate `marks` (must be character vector of length >=1, must not be `NULL`)

Reply:
If the vector is length 1, then also test that it's a valid value

Reply:
Input validation for `marks` has been added (new testthat tests added as well).